src/Cake.Svn/SvnAliases.Info.cs

```diff
@@ -0,0 +1,255 @@
+        /// <code>
+        /// <![CDATA[
+        ///     var isWorkingCopy = IsWorkingCopy(@"C:\project\src\newfolder\");
+        ///
+        ///     Verbose("The directory is a working copy: {0}", isWorkingCopy);
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Info")]
+        [CakeNamespaceImport("Cake.Svn.Info")]
+        public static bool IsWorkingCopy(this ICakeContext context, DirectoryPath directoryPath)
+        {
+            var svnInformation = new SvnInfo(context.Environment, SvnClientFactoryMethod);
+
+            return svnInformation.IsWorkingCopy(directoryPath);
+        }
+
+        /// <summary>
+        /// Check if the <paramref name="filePath"/> is controlled by Subversion.
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <param name="filePath">The file.</param>
+        /// <returns>
+        /// <c>true</c> if the the file <paramref name="filePath"/> is controlled by Subversion. 
+        /// Otherwise <c>false</c> will be returned.
+        /// </returns>
+        /// <example>
+        /// Check if the <paramref name="filePath"/> is controlled by Subversion.
+        /// <code>
+        /// <![CDATA[
+        ///     var isWorkingCopy = IsWorkingCopy(@"C:\project\src\newfolder\newfile.cs");
+        ///
+        ///     Verbose("The file is a working copy: {0}", IsWorkingCopy);
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Info")]
+        [CakeNamespaceImport("Cake.Svn.Info")]
+        public static bool IsWorkingCopy(this ICakeContext context, FilePath filePath)
+        {
+            var svnInformation = new SvnInfo(context.Environment, SvnClientFactoryMethod);
+
+            return svnInformation.IsWorkingCopy(filePath);
+        }
+
+        /// <summary>
+        /// Gets Subversion information about the directory at <paramref name="directoryPath"/>.
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <param name="directoryPath">The directory.</param>
+        /// <returns>The Subversion information.</returns>
+        /// <example>
+        /// <para>Gets Subversion information about the directory at <paramref name="directoryPath"/>.</para>
+        /// <code>
+        /// <![CDATA[
+        ///     var localDirectoryInfo = GetDirectoryInfo(@"C:\project\src\");
+        ///
+        ///     foreach(var svnInfoResult in localDirectoryInfo)
+        ///     {
+        ///         Verbose("Path: {0}", svnInfoResult.Path);
+        ///     }
+        /// ]]>
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Info")]
+        [CakeNamespaceImport("Cake.Svn.Info")]
+        public static IEnumerable<SvnInfoResult> GetDirectoryInfo(this ICakeContext context, DirectoryPath directoryPath)
```

Comment:
To avoid potential naming conflicts with other addins I suggest to name this `GetSvnDirectoryInfo`